test/SparkLendMainnetIntegration.t.sol

```diff
@@ -0,0 +1,287 @@
+                variableRateSlope1Spread: -0.004e27,  // -0.4% spread
+                variableRateSlope2:       0.2e27
+            });
+        IDefaultInterestRateStrategy prevStrategy
+            = IDefaultInterestRateStrategy(USDC_USDT_IRM);
+
+        assertEq(strategy.OPTIMAL_USAGE_RATIO(),       prevStrategy.OPTIMAL_USAGE_RATIO());
+        assertEq(strategy.MAX_EXCESS_USAGE_RATIO(),    prevStrategy.MAX_EXCESS_USAGE_RATIO());
+        assertEq(strategy.getBaseVariableBorrowRate(), prevStrategy.getBaseVariableBorrowRate());
+        assertEq(strategy.getVariableRateSlope1(),     prevStrategy.getVariableRateSlope1());
+        assertEq(strategy.getVariableRateSlope2(),     prevStrategy.getVariableRateSlope2());
+        assertEq(strategy.getMaxVariableBorrowRate(),  prevStrategy.getMaxVariableBorrowRate());
+
+        _triggerUpdate(USDC);
+        _triggerUpdate(USDT);
+
+        uint256 currentRateUSDC = 0.042370487857489861390961744e27;
+        uint256 currentRateUSDT = 0.042533501480957788642828711e27;
+
+        assertEq(_getBorrowRate(USDC), currentRateUSDC, "before: USDC mismatch");
+        assertEq(_getBorrowRate(USDT), currentRateUSDT, "before: USDT mismatch");
+
+        vm.startPrank(ADMIN);
+        configurator.setReserveInterestRateStrategyAddress(
+            USDC,
+            address(strategy)
+        );
+        configurator.setReserveInterestRateStrategyAddress(
+            USDT,
+            address(strategy)
+        );
+        vm.stopPrank();
+
+        _triggerUpdate(USDC);
+        _triggerUpdate(USDT);
+
+        // Should be no change
+        assertEq(_getBorrowRate(USDC), currentRateUSDC, "after1: USDC mismatch");
+        assertEq(_getBorrowRate(USDT), currentRateUSDT, "after1: USDT mismatch");
+
+        // Test with different parameters
+        strategy = new RateTargetKinkInterestRateStrategy({
+            provider:                 poolAddressesProvider,
+            rateSource:               address(new PotRateSource(POT)),
+            optimalUsageRatio:        0.95e27,
+            baseVariableBorrowRate:   0,
+            variableRateSlope1Spread: -0.002e27,  // -0.2% spread
+            variableRateSlope2:       0.2e27
+        });
+
+        vm.startPrank(ADMIN);
+        configurator.setReserveInterestRateStrategyAddress(
+            USDC,
+            address(strategy)
+        );
+        configurator.setReserveInterestRateStrategyAddress(
+            USDT,
+            address(strategy)
+        );
+        vm.stopPrank();
+
+        _triggerUpdate(USDC);
+        _triggerUpdate(USDT);
+
+        // Almost +0.2%, but utilization is slightly below the kink point
+        assertEq(_getBorrowRate(USDC), currentRateUSDC + 0.001891955013226013824859422e27, "after2: USDC mismatch");
+        assertEq(_getBorrowRate(USDT), currentRateUSDT + 0.001899234005225860815063204e27, "after2: USDT mismatch");
+    }
+
+    // TODO add capped oracles for WBTC (need to import the combining contract first)
```

Comment:
TODO: Add

Reply:
What does combining contract mean here?

Reply:
Fixed and moved comment to top of file like you normally do. WBTC oracle is a combination of WBTC/BTC * BTC/USD oracles. There is a contract that BGD made to combine these. We need to import that into this repo to support a capped WBTC/BTC price, but it needs to convert to USD.

Reply:
added colon :